pkg/internal/blockutil.go

```diff
@@ -0,0 +1,237 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	ExecCommand          = exec.Command
+	FilePathGlob         = filepath.Glob
+	FilePathEvalSymLinks = filepath.EvalSymlinks
+	mountFile            = "/proc/1/mountinfo"
+)
+
+const (
+	// StateSuspended is a possible value of BlockDevice.State
+	StateSuspended = "suspended"
+	// DiskByIDDir is the path for symlinks to the device by id.
+	DiskByIDDir = "/dev/disk/by-id/"
+)
+
+// BlockDevice is the a block device as output by lsblk.
+// All the fields are lsblk columns.
+type BlockDevice struct {
+	Name   string `json:"name"`
+	KName  string `json:"kname"`
+	Type   string `json:"type"`
+	Model  string `json:"model,omitempty"`
+	Vendor string `json:"vendor,omitempty"`
+	State  string `json:"state,omitempty"`
+	FSType string `json:"fsType"`
+	Size   string `json:"size"`
+	// Children   []BlockDevice `json:"children,omitempty"`
+	Rotational string `json:"rota"`
+	ReadOnly   string `json:"ro,omitempty"`
+	Removable  string `json:"rm,omitempty"`
+	PathByID   string `json:"pathByID,omitempty"`
+	Serial     string `json:"serial,omitempty"`
+	PartLabel  string `json:"partLabel,omitempty"`
+}
+
+// ListBlockDevices using the lsblk command
+func ListBlockDevices() ([]BlockDevice, []string, error) {
+	// var output bytes.Buffer
+	var blockDevices []BlockDevice
+
+	deviceFSMap, err := GetDeviceFSMap()
```

Comment:
lsblk seems to have an FSTYPE output option. Does this need to be done using blkid?
Can be looked into in a separate PR.

Reply:
There were some issues around that.  Need to find the BZ around this.